src/distance.hpp

```diff
@@ -0,0 +1,185 @@
+         const Snarl* snarl1, const Snarl* snarl2, pos_t& pos1, pos_t& pos2);
+  
+    protected:
+    void printSelf();
+    class SnarlDistances {
+        
+        /* Stores distance information for nodes in a snarl.
+           visitToIndex maps each visit (node_id, negative if reverse) to an int
+           distances stores all the distance bewteen each pair of visits in a 
+           snarl
+        */
+
+        public:
+        
+            //Constructor
+            SnarlDistances(unordered_set<pair<id_t, bool>>& allNodes, 
+                          pair<id_t, bool> start, pair<id_t, bool> end);
+           
+
+            //Distance between beginning of node start and beginning of node end
+            int64_t snarlDistance(pair<id_t, bool> start, pair<id_t, bool> end);
+
+ 
+            //Distance between end of node start and beginning of node end
+            int64_t snarlDistanceShort(VG* graph,NetGraph* ng,
+                     pair<id_t, bool> start, pair<id_t, bool> end); 
+
+            //Add the distance from start to end to the index
+            void insertDistance(pair<id_t, bool> start, pair<id_t, bool> curr, 
+                     int64_t dist);
+             
+            //Length of a node
+            int64_t nodeLength(VG*graph, NetGraph* ng,  pair<id_t, bool> node);
+        
+            //Total length of the snarl
+            int64_t snarlLength();
+
+            /*Given distances from a position to either end of a node, find the
+              shortest distance from that position to the start and end nodes of
+              the snarl
+            */
+            pair<int64_t, int64_t> distToEnds(id_t node, bool rev, int64_t distL,
+                                                                int64_t distR);
+
+            void printSelf();
+
+        protected:
+
+            //Maps node to index to get its distance
+            unordered_map< pair<id_t, bool>, size_t> visitToIndex;
+ 
+             //Store the distance between every pair nodes, -1 indicates no path
+             //For child snarls that are unary or only connected to one node
+             //in the snarl, distances between that node leaving the snarl
+             //and any other node is -1
+            vector<int64_t> distances;
+
+            //ID of the first node in the snarl, also key for distance index 
+            pair<id_t, bool> snarlStart;
+ 
+           //End facing out of snarl
+            pair<id_t, bool> snarlEnd;           
+
+            //Total length of the snarl- start to end plus length of end
+            int64_t length; 
+
+            //The index into distances for distance start->end
+            size_t index(pair<id_t, bool> start, pair<id_t, bool> end);
+
+        private: 
+
+            int64_t snarlDistanceShortHelp(VG* graph,NetGraph* ng,
+                     pair<id_t, bool> start, pair<id_t, bool> end); 
+
+
+        friend class DistanceIndex;
+        friend class TestDistanceIndex;
+        };
```

Comment:
Small thing, but this is an indentation mismatch--makes it a little hard to read.